x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/DataFrameDataExtractorFactory.java

```diff
@@ -33,6 +39,18 @@
     private static final List<String> IGNORE_FIELDS = Arrays.asList("_id", "_field_names", "_index", "_parent", "_routing", "_seq_no",
         "_source", "_type", "_uid", "_version", "_feature", "_ignored");
 
+    /**
+     * The types supported by data frames
+     */
+    private static final Set<String> COMPATIBLE_FIELD_TYPES;
+
+    static {
+        COMPATIBLE_FIELD_TYPES = Stream.of(NumberFieldMapper.NumberType.values())
+            .map(NumberFieldMapper.NumberType::typeName)
+            .collect(Collectors.toSet());
+        COMPATIBLE_FIELD_TYPES.add("scaled_float"); // have to add manually since scaled_float is in a module
```

Comment:
It would be nice if `COMPATIBLE_FIELD_TYPES` was assigned a `Collections.unmodifiableSet`.  So the static block could build up a temporary set then assign an unmodifiable version to `COMPATIBLE_FIELD_TYPES` at the end.

Reply:
Done